cmake/modules/snippets.cmake

```diff
@@ -65,6 +65,15 @@ function(zephyr_process_snippets)
     file(REAL_PATH ${snippet_dir} real_snippet_dir)
     list(APPEND real_snippet_root ${real_snippet_dir})
   endforeach()
+
+  # Search through all modules for any that have a snippets directory
+  foreach(module_name ${ZEPHYR_MODULE_NAMES})
+    string(TOUPPER "${module_name}" module_name)
+    if(EXISTS "${ZEPHYR_${module_name}_MODULE_DIR}/snippets")
+      list(APPEND real_snippet_root "${ZEPHYR_${module_name}_MODULE_DIR}")
+    endif()
+  endforeach()
```

Comment:
we should not automatically loop all modules for a snippet root.

The `zephyr/module.yml` file of the module should instead use:
```
build:
  settings:
    snippet_root: <relative-path-to-root>
```
Ref: https://docs.zephyrproject.org/latest/develop/modules.html#build-settings
> snippet_root: Contains additional snippets that are available for use. These snippets must be defined in snippet.yml files underneath the <snippet_root>/snippets folder. 

```suggestion
```